cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -158,6 +160,13 @@ func WithStartTime(startTime pcommon.Timestamp) metricBuilderOption {
 	}
 }
 
+// WithBuildInfo sets BuildInfo on the metrics builder.
+func WithBuildInfo(info component.BuildInfo)  metricBuilderOption {
+	return func(mb *MetricsBuilder) {
+		mb.buildInfo = info
+	}
+}
```

Comment:
Do we need to expose it until there is a need?

Reply:
Oh, I thought this is needed to override the default... Can't we set it right in the generated code?

Reply:
My reasoning is that it's a bit inconsistent: we hardcode the scope name, but provide an option to customize the version. I would like to avoid providing that option until there is a need for that

Reply:
Are you suggesting passing BuildInfo as a parameter to NewMetricsBuilder instead?

Reply:
Yes, I believe we should pass it as the another argument to `NewMetricsBuilder` to have it set consistently. Let me know what do you think

Reply:
@dmitryax PTAL, i've updated `NewMetricsBuilder`, this made the change much bigger, as I had to change its usage across all components